src/modules/twap/hooks/useTwapWarningsContext.ts

```diff
@@ -16,7 +16,7 @@ export function useTwapWarningsContext(): TwapWarningsContext {
   return useMemo(() => {
     // TODO: bind to settings
     const expertMode = false
-    const canTrade = !primaryFormValidation
+    const canTrade = !primaryFormValidation || primaryFormValidation >= TradeFormValidation.ExpertApproveAndSwap
```

Comment:
It's an ENUM of numbers and the order of parameters matters. I'm not sure that it's a reliable solution, but it works...

<img width="278" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/389cbc7a-25a2-4044-8b7f-7d5e60e64b5d">

Reply:
Hmm, not very found of this as we could for instance re-order the enum alphabetically without noticing it would break this, TS wouldn't warn about it.

I would prefer to create a list of allowed values, such as:

`valid_states_or_whatever = [ExpertApproveAndSwap, ApproveAndSwap, ApproveRequired]`

Reply:
Agree, looks more reliable solution.
Fixed it here: https://github.com/cowprotocol/cowswap/pull/2764/commits/711a397bb0f5456480b589a19e88870f94b94faf

And sorry for that, sometimes I feel the urge to create strange new solutions :)

Reply:
Thank you!